hazelcast/src/main/java/com/hazelcast/internal/partition/operation/ReplicaSyncRequest.java

```diff
@@ -124,16 +148,6 @@ private boolean preCheckReplicaSync(NodeEngineImpl nodeEngine, int partitionId,
             return false;
         }
 
-        if (currentVersion == 0) {
```

Comment:
If replica version is `0`, then service (most probably) won't create a replication operation, because it will not have any data for that replica. Even if service creates an operation, it will be just empty, won't contain any data.